diff --git a/resources/workbenchModes/agent-sessions.code-workbench-mode b/resources/workbenchModes/agent-sessions.code-workbench-mode
index 0fd107b02de81..7aa37f33888ec 100644
--- a/resources/workbenchModes/agent-sessions.code-workbench-mode
+++ b/resources/workbenchModes/agent-sessions.code-workbench-mode
@@ -19,6 +19,7 @@
 		"workbench.sideBar.location": "right",
 		"workbench.statusBar.visible": false,
 		"workbench.secondarySideBar.forceMaximized": true,
+		"workbench.secondarySideBar.defaultVisibility": "maximized",
 		"workbench.startupEditor": "none",
 		"workbench.tips.enabled": false,
 		"workbench.layoutControl.type": "toggles",
diff --git a/src/vs/workbench/browser/layout.ts b/src/vs/workbench/browser/layout.ts
index fc7da781979d5..abd34a528b5ac 100644
--- a/src/vs/workbench/browser/layout.ts
+++ b/src/vs/workbench/browser/layout.ts
@@ -2981,11 +2981,9 @@ class LayoutStateModel extends Disposable {
 	private applyOverrides(configuration: ILayoutStateLoadConfiguration): void {
 
 		// Auxiliary bar: Maximized settings
-		const auxiliaryBarForceMaximized = this.configurationService.getValue(WorkbenchLayoutSettings.AUXILIARYBAR_FORCE_MAXIMIZED);
-		if (this.isNew[StorageScope.WORKSPACE] || auxiliaryBarForceMaximized) {
+		if (this.isNew[StorageScope.WORKSPACE]) {
 			const defaultAuxiliaryBarVisibility = this.configurationService.getValue(WorkbenchLayoutSettings.AUXILIARYBAR_DEFAULT_VISIBILITY);
 			if (
-				auxiliaryBarForceMaximized ||
 				defaultAuxiliaryBarVisibility === 'maximized' ||
 				(defaultAuxiliaryBarVisibility === 'maximizedInWorkspace' && this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY)
 			) {
