diff --git a/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts b/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts
index 3e5d9212f1881..3fe0121baae16 100644
--- a/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts
+++ b/src/vs/workbench/contrib/terminalContrib/suggest/browser/terminal.suggest.contribution.ts
@@ -232,7 +232,7 @@ class TerminalSuggestContribution extends DisposableStore implements ITerminalCo
 		const container = this._resolveAddonContainer(xtermElement);
 		addon.setContainerWithOverflow(container);
 		// eslint-disable-next-line no-restricted-syntax
-		const screenElement = xtermElement.querySelector('.xterm-screen');
+		const screenElement = xtermElement?.querySelector('.xterm-screen');
 		if (dom.isHTMLElement(screenElement)) {
 			addon.setScreen(screenElement);
 		}
