diff --git a/src/vs/workbench/contrib/chat/browser/widgetHosts/viewPane/chatViewPane.ts b/src/vs/workbench/contrib/chat/browser/widgetHosts/viewPane/chatViewPane.ts
index bf4790ed7cf73..b65a428df20e9 100644
--- a/src/vs/workbench/contrib/chat/browser/widgetHosts/viewPane/chatViewPane.ts
+++ b/src/vs/workbench/contrib/chat/browser/widgetHosts/viewPane/chatViewPane.ts
@@ -539,6 +539,7 @@ export class ChatViewPane extends ViewPane implements IViewWelcomeDelegate {
 				enableImplicitContext: true,
 				enableWorkingSet: 'explicit',
 				supportsChangingModes: true,
+				dndContainer: parent,
 			},
 			{
 				listForeground: SIDE_BAR_FOREGROUND,
