diff --git a/test/smoke/src/areas/chat/chatAnonymous.test.ts b/test/smoke/src/areas/chat/chatAnonymous.test.ts
index 8ee9bc1cc2cd7..520292a96d4a7 100644
--- a/test/smoke/src/areas/chat/chatAnonymous.test.ts
+++ b/test/smoke/src/areas/chat/chatAnonymous.test.ts
@@ -7,7 +7,7 @@ import { Application, Logger } from '../../../../automation';
 import { installAllHandlers } from '../../utils';
 
 export function setup(logger: Logger) {
-	describe('Chat Anonymous', () => {
+	describe.skip('Chat Anonymous', () => {
 
 		// Shared before/after handling
 		installAllHandlers(logger);
@@ -31,7 +31,7 @@ export function setup(logger: Logger) {
 			await app.workbench.chat.waitForResponse();
 
 			// Wait for model name to appear in footer
-			// await app.workbench.chat.waitForModelInFooter();
+			await app.workbench.chat.waitForModelInFooter();
 		});
 	});
 }
