diff --git a/src/vs/workbench/contrib/chat/browser/actions/chatNewActions.ts b/src/vs/workbench/contrib/chat/browser/actions/chatNewActions.ts
index 271e82b2baf41..cdc8b4e285826 100644
--- a/src/vs/workbench/contrib/chat/browser/actions/chatNewActions.ts
+++ b/src/vs/workbench/contrib/chat/browser/actions/chatNewActions.ts
@@ -205,6 +205,16 @@ export function registerNewChatActions() {
 		order: 1
 	});
 
+	MenuRegistry.appendMenuItem(MenuId.ChatTitleBarMenu, {
+		command: {
+			id: ACTION_ID_NEW_CHAT,
+			title: localize2('chat.newEdits.label', "New Chat"),
+		},
+		when: ChatContextKeys.enabled,
+		group: 'b_new',
+		order: -1,
+	});
+
 	registerAction2(class UndoChatEditInteractionAction extends EditingSessionAction {
 		constructor() {
 			super({
